cmd/checkpoint/main.go

```diff
@@ -311,6 +334,28 @@ func getLocalParentPods() (map[string]*v1.Pod, error) {
 	return podListToParentPods(&podList), nil
 }
 
+// getLocalRunningPods uses the /runningpods/ kubelet api to retrieve the local container runtime pod state
+func getLocalRunningPods() (map[string]*v1.Pod, error) {
+	// TODO(aaron): The kubelet api is currently secured by a self-signed cert. We should update this to actually verify at some point
```

Comment:
So a couple points on this.

- Can the checkpointer manifests include secrets? We could include the CA later there if we needed to.
- One of the options for kubelet API auth(n/z) is the "token webhook" mode. Which let's you use things like service account tokens to query the kubelet API, but it requires the kubelet to be able to query the API server to determine if the token is valid and has the correct permissions. (see: `--authentication-token-webhook` and `--authorization-mode=Webhook`)
- This means we'll have to use client certs here if we ever want to do kubelet auth.

This isn't show stopping but something we should be aware of.